### ISSUE TITLE
Web component: Link list section - "Lorem ipsum dolor sit amet" - missing heading tag. 

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#4844]
## Detailed description

Issue: 
"Lorem ipsum dolor sit amet" - This text visually looks prominent like a heading. But not given heading tag

Expected:
If this is a heading, Provide appropriate heading tag

## Additional information

![image.png](https://images.zenhubusercontent.com/5e21a7b470118a498429a20e/e57b5dae-4133-4385-9005-fd7b03349e4a)

**Comment by asudoh** [comment]
The fix should be adding `heading` role, similar to what `<dds-card-heading>` does.
